nbody_gravitational/NBodyGravitational.jl

```diff
@@ -0,0 +1,173 @@
+module NBodyGravitational
+
+using DiffEqBase, DifferentialEquations, StaticArrays, Plots, StaticArrays
+
+
+grav_const = 6.673e-11
+
+#=
+    Represents a body/particle in an N-body gravitational problem
+    m - mass
+    r - initial position
+    v - initial velocity
+=#
+struct MassBody{mType, cType}
+    m :: mType
+    r :: Vector{cType}
```

Comment:
What about using static arrays to make this a bits value type?

Reply:
Yes, good question. I used to use `SVector` for these data (`r` and `v`), but if there are a lot of bodies, they take quite big place in the stack memory. Say, 1000 bodies and 1000x2x3 elements to store. Also, `MassBody` is just an intermediate container and doesn't really play role in actual calculation of the differential problem. That is why I use mere `Vector`s.

Reply:
A `Vector{SMatrix}` of length 1000 just has the same memory layout as a 3D tensor. It wouldn't have issues with stack memory. You'd only have issues here if it had hundreds of dimensions.

Reply:
Ok I changed to SVectors -  solution is evaluated quicker.